src/core/meshcop/joiner.cpp

```diff
@@ -129,7 +129,10 @@ otError Joiner::Start(const char *     aPskd,
     if (error != OT_ERROR_NONE)
     {
         otLogInfoMeshCoP("Error %s while starting joiner", otThreadErrorToString(error));
-        FreeJoinerFinalizeMessage();
+        if (mState == OT_JOINER_STATE_IDLE)
```

Comment:
How about initializing `error` as `OT_ERROR_NONE` at the begining?

Line#101 has the status check already, here not necessary to check again if `error` properly initiazlied.